monitoring/bin/change_grafana_admin_password.sh

```diff
@@ -3,140 +3,145 @@
-      POS_PARMS="$POS_PARMS $1"
-      shift
-      ;;
-  esac
+        POS_PARMS="$POS_PARMS $1"
+        shift
+        ;;
+    esac
 done
 
 # Convert namespace and tenant to all lower-case
-tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
-tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+tenantNS=$(echo "$tenantNS" | tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant" | tr '[:upper:]' '[:lower:]')
 
 # Check for parameters - set with cluster or tenant
 if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
-  cluster="true"
-  namespace=$MON_NS
-  grafanaInstance="v4m-grafana"
+    cluster="true"
+    namespace=$MON_NS
+    grafanaInstance="v4m-grafana"
 elif [ -n "$tenantNS" ] && [ -n "$tenant" ]; then
-  cluster="false"
-  namespace=$tenantNS
-  grafanaInstance="v4m-grafana-${tenant}"
+    cluster="false"
+    namespace=$tenantNS
+    grafanaInstance="v4m-grafana-${tenant}"
 else
-  log_error "Both a [NAMESPACE] and a [TENANT] are required in order to change the Grafana admin password.";
-  exit 1
+    log_error "Both a [NAMESPACE] and a [TENANT] are required in order to change the Grafana admin password."
+    exit 1
 fi
 
 # Check and make sure that a password was provided.
-if [ -z $password ]; then
-  log_error "A value for parameter [PASSWORD] has not been provided"
-  exit 1
-fi 
+if [ -z "$password" ]; then
+    log_error "A value for parameter [PASSWORD] has not been provided"
+    exit 1
+fi
 
 if [ "$cluster" == "true" ]; then
-  grafanaPod="$(kubectl get pods -n $namespace -l app.kubernetes.io/name=grafana --template='{{range .items}}{{.metadata.name}}{{end}}')"
+    grafanaPod=$(kubectl get pods -n "$namespace" -l app.kubernetes.io/name=grafana --template='{{range .items}}{{.metadata.name}}{{end}}')
 else
-  grafanaPod="$(kubectl get pods -n $namespace -l app.kubernetes.io/name=grafana -l app.kubernetes.io/instance=$grafanaInstance --template='{{range .items}}{{.metadata.name}}{{end}}')"
+    grafanaPod=$(kubectl get pods -n "$namespace" -l app.kubernetes.io/name=grafana -l app.kubernetes.io/instance="$grafanaInstance" --template='{{range .items}}{{.metadata.name}}{{end}}')
 fi
 
 # Error out if a Grafana pod has not been found
-if [ -z $grafanaPod ]; then
-  log_error "Unable to update Grafana password."
-  log_error "No Grafana pods were available to update"
-  exit 1
+if [ -z "$grafanaPod" ]; then
+    log_error "Unable to update Grafana password."
+    log_error "No Grafana pods were available to update"
+    exit 1
 fi
 
 # Changes the admin password using the Grafana CLI
 log_info "Updating the admin password using Grafana CLI"
-kubectl exec -n $namespace $grafanaPod -c grafana -- bin/grafana-cli admin reset-admin-password $password
-
-# Exit out of the script if the Grafana CLI call fails
-if (( $? != 0 )); then
-  log_error "An error occurred when updating the password"
-  exit 1
+if ! kubectl exec -n "$namespace" "$grafanaPod" -c grafana -- bin/grafana-cli admin reset-admin-password "$password"; then
```

Comment:
Interesting that this is preferred...definitely looks a bit odd.

Reply:
I actually prefer the way it looks but if you don't we can change it back, I'm not that stuck on it

Reply:
No, I don't think there's a need to revert this, I can get used to it.  I was going to ask about whether we need/want to redirect stderr/stdout but I guess any messages generated by the command could be useful...and, as I recall, since it is part of an IF statement, the exit-on-error (set -e) processing won't kick in.